TombLib/TombLib/LevelData/RoomGeometry.cs

```diff
@@ -1502,7 +1497,7 @@ private void AddVerticalFaces(Room room, int x, int z, FaceDirection faceDirecti
                 ZB = zB,
             };
 
-            if (_legacy)
+            if (useLegacyCode)
```

Comment:
Maybe you can use a delegate instead?
Provide a legacy `AddXYZFace` method and a new `AddXYZFace`.
And set the delegate accordingly?